NEWS.md

```diff
@@ -2,7 +2,19 @@ MXNet Change Log
 ================
 ## 1.2.1
 ### Deprecations
-- An incorrect [usage](https://github.com/apache/incubator-mxnet/issues/11091) of `save_params` was advertised in the gluon book which led to MXNet users depending on the incorrect usage and developing a hack around it. A change was made to the internal structure of the `.params` file saved by `save_params` to resolve a bug. This led to user scripts with the above mentioned hack to break. To fix this, `save_params` and `load_params` APIs have been reverted to previous format and marked as deprecated. New APIs: `save_parameters` and `load_parameters` have been added for the new format. All scripts to save and load parameters for a Gluon model should now use the new API for `save_parameters` and `load_parameters`. If your model is hybridizable and you want to export a serialized structure of the model as well as parameters you need to use the `export` API and the newly added `imports` API instead of `save_params` and `load_params` API. For more details, Please see: [issue](https://github.com/apache/incubator-mxnet/issues/11091), [PR](https://github.com/apache/incubator-mxnet/pull/11127).
+The [usage](https://github.com/apache/incubator-mxnet/issues/11091) of `save_params` described in the [gluon book](https://github.com/zackchase/mxnet-the-straight-dope/blob/master/chapter07_distributed-learning/hybridize.ipynb) did not reflect the intended usage of the API and led MXNet users to depend on the unintended usage of `save_params` and `load_params`. In 1.2.0 release an internal bug fix was made which broke the unintended usage use case and users scripts.
+To correct the API change, the behavior of `save_params`/`load_params` API has been reverted to the behavior of MXNet v1.1.0 in v1.2.1. The intended and correct use are now supported with the new APIs `save_parameters` and `load_parameters`.
+With v1.2.1, usage of `save_params` and `load_params` APIs will resume their former functionality and a deprecation warning will appear.
+All scripts to save and load parameters for a Gluon model should use the new APIs: `save_parameters` and `load_parameters`. If your model is hybridizable and you want to export a serialized structure of the model as well as parameters you should migrate your code to use `export` API and the newly added `imports` API instead of `save_params` and `load_params` API. Please refer to the [Saving and Loading Gluon Models Tutorial](https://mxnet.incubator.apache.org/tutorials/gluon/save_load_params.html) for more information.
```

Comment:
I disagree with the assessment above.

Reply:
can you please pinpoint which part is not factual in your opinion?

Reply:
The tutorial that NEWS.md points to for "more information" only mentions save_params() and load_params().

Reply:
For example, load_params was not reverted to the behavior of 1.1.0 in 1.2.1. I think Anirudh should already be aware of it.

Reply:
The statement talks about `save_params`, `load_params` in conjunction. I think the revert statement makes sense for our users.
From the user point of view, if they were using `save_params`/`load_params` to save their model architecture and params, to load it back later into Symbolblock, they know that the their codes which broke in 1.2 will not break in 1.2.1. They are also informed that they should use export and imports API. For users who used it to create layers without namescope in v1.2.0, they know that this feature didn't work with save_params and load_params in v1.1.0 so they will switch to save_parameters and load_parameters.

Reply:
@safrooze thanks for pointing! I have changed the link. Also opened a PR to fix the doc for master branch.

Reply:
@anirudh2290 I'd suggest to opt for correct description of the behavior (like you just did) over incorrect description of what has been done.

Reply:
@szha how are you suggesting that the text be changed ?

Reply:
ping @szha , @Roshrini has opened PR to cherry pick warning message changes : https://github.com/apache/incubator-mxnet/pull/11548. Can you please call out specific concerns with the release notes.

Reply:
I already did above. Please update.

Reply:
I have already explained that (`save_params`\`load_params`) is in conjunction and why the revert statement makes sense to our users.

Reply:
@anirudh2290 you asked for my specific concerns, so I explained that my concern is false statement, and I offered alternative that doesn't require false statement.

Seems weird to me that the justification for making false statement is because it "makes sense to our users".

Reply:
How is it false that the behavior of `save_params`/`load_params` API has been reverted to the behavior of 1.1.0 ? Does the line say that `load_params` API has been reverted to 1.1.0 ?

Reply:
The code for `load_params` (and its behavior) wasn't reverted to what's in 1.1.0, for the best compatibility.

Maybe I misunderstood what you were attempting to say with the distinction between "API has been reverted to the behavior of 1.1.0" v.s. "API has been reverted to 1.1.0". I was treating them as equivalent in this case. So please bear with me :)

My suggestion would be to provide explanation as faithful to the code as possible, so that the explanation would always be true regardless of the users' level of understanding. Besides that, you may add whatever you think would benefit the users as you see fit.